packages/ast-tooling/index.ts

```diff
@@ -173,9 +173,42 @@ export function parseJson(content: string) {
     return fleece.evaluate(content);
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function serializeJson(originalInput: string, data: any) {
+export function serializeJson(originalInput: string, data: unknown) {
     // some of the files we need to process contain comments. The default
     // node JSON.parse fails parsing those comments.
-    return fleece.patch(originalInput, data);
+    const spaces = guessIndentString(originalInput);
+    return fleece.stringify(data, { spaces });
```

Comment:
This brings up a completely different topic that we had previously, but that does not apply at this point. 

Let's assume we have a `tsconfig.json` like this (created by `create-svelte`)
```json
{
	"extends": "./.svelte-kit/tsconfig.json",
	"compilerOptions": {
		"allowJs": true,
		"checkJs": true,
		"esModuleInterop": true,
		"forceConsistentCasingInFileNames": true,
		"resolveJsonModule": true,
		"skipLibCheck": true,
		"sourceMap": true,
		"strict": true,
		"moduleResolution": "bundler"
	}
	// Path aliases are handled by https://kit.svelte.dev/docs/configuration#alias
	// except $lib which is handled by https://kit.svelte.dev/docs/configuration#files
	//
	// If you want to overwrite includes/excludes, make sure to copy over the relevant includes/excludes
	// from the referenced tsconfig.json - TypeScript does not merge them in
}
```

With the approach we would now implement the comment would disappear, because the comments are not part of the JSON object. Currently we don't have that case active anymore, because we removed the routify? adder recently where this usecase came from. 

In order to get the best out of both worlds we should maybe introduce a new file type `json-comments` here 

https://github.com/svelte-add/svelte-add/blob/2f9a963fa826864c8207582b2f8fc3b83a893c44/packages/core/files/processors.ts#L56

and introduce some different handling, as I'm sure such a thing will happen in the future. On the other hand we could also leave it open for the future.

Reply:
We will just need to remember that in the future! But ok for now.